src/utils/editor/source-search.js

```diff
@@ -208,6 +233,46 @@ function searchNext(ctx, rev, query, newQuery, modifiers) {
   return nextMatch;
 }
 
+function findNextOnLine(ctx, rev, query, newQuery, modifiers, line) {
+  const { cm, ed } = ctx;
+  let nextMatch;
+  cm.operation(function() {
+    const state = getSearchState(cm, query);
+    state.posFrom.ch = 0;
+    state.posFrom.line = line - 1;
+    state.posTo.ch = null;
+    state.posTo.line = null;
+    const pos = getCursorPos(newQuery, rev, state);
+
+    if (!state.query) {
+      return;
+    }
+
+    let cursor = getSearchCursor(cm, state.query, pos, modifiers);
+    const location = rev
+      ? { line: cm.lastLine(), ch: null }
+      : { line: cm.firstLine(), ch: 0 };
+
+    if (!cursor.find(rev) && state.query) {
+      cursor = getSearchCursor(cm, state.query, location, modifiers);
+      if (!cursor.find(rev)) {
+        return;
+      }
+    }
+
+    // We don't want to jump the editor
+    // when we're selecting text
+    if (!cm.state.selectingText) {
+      ed.alignLine(cursor.from().line, "center");
+      cm.setSelection(cursor.from(), cursor.to());
+    }
+
+    nextMatch = { from: cursor.from(), to: cursor.to() };
+  });
+
+  return nextMatch;
+}
```

Comment:
how is this different from searchNext? could we re-use `searchNext` are there are parts that we want to share?

Reply:
Yeah, so I guess `searchNext` needs to actually find the next instance of the query to full highlight, while `findNextForSearch` would already know the starting `line` and `ch`, so it doesn't need to do a bunch of logic to find the starting location.